lib/kubernetes-deploy/kubernetes_resource/horizontal_pod_autoscaler.rb

```diff
@@ -0,0 +1,25 @@
+# frozen_string_literal: true
+module KubernetesDeploy
+  class HorizontalPodAutoscaler < KubernetesResource
+    PRUNABLE = true
+    TIMEOUT = 30.seconds
+
+    def deploy_succeeded?
+      conditions = @instance_data.dig("status", "conditions") || []
+      able_to_scale = conditions.detect { |c| c["type"] == "AbleToScale" } || {}
+      able_to_scale["status"] == "True"
+    end
+
+    def deploy_failed?
+      !exists?
+    end
+
+    def timeout_message
+      UNUSUAL_FAILURE_MESSAGE
```

Comment:
I'm not sure this message (`It is very unusual for this resource type to fail to deploy. Please try the deploy again. If that new deploy also fails, contact your cluster administrator.`) is fitting, since HPAs have legit failure mode (unlike say configmaps).